pkg/kademlia/routing.go

```diff
@@ -140,6 +140,27 @@ func (rt *RoutingTable) FindNear(id storj.NodeID, limit int) (nodes []*pb.Node,
 	return nodes, nil
 }
 
+// UpdateSelf updates a node on the routing table
+func (rt *RoutingTable) UpdateSelf(node *pb.Node) error {
+	if node.Id != rt.Local().Id {
+		return RoutingErr.New("self does not have a matching node id")
+	}
+
+	rt.mutex.Lock()
```

Comment:
curious, why not just keep the mutex ?

Reply:
I think I'm missing something because the mutex lock is there